source/common/http/http2/codec_impl.cc

```diff
@@ -1027,10 +1000,9 @@ Status ConnectionImpl::sendPendingFrames() {
       return nghttp2_callback_status_;
     }
 
-    // The frame flood error should set the nghttp2_callback_status_ error, and return at the
-    // statement above.
-    ASSERT(outbound_frames_ <= max_outbound_frames_ &&
-           outbound_control_frames_ <= max_outbound_control_frames_);
+    // Protocol constrain violations should set the nghttp2_callback_status_ error, and return at
+    // the statement above.
+    ASSERT(protocol_constraints_.status().ok());
```

Comment:
Is it possible to fail this ASSERT in the upstream direction?  Should you be calling ASSERT(checkProtocolConstraintsStatus()); instead?

Reply:
It should not be anymore. Both inbound and outbound frame limits are only checked in the server codec now.

Reply:
Should not be possible now, since the flood checks that can flip status are only done by the server codec.